File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.kt
class Solution {

    fun solution(video_len: String, pos: String, op_start: String, op_end: String, commands: Array<String>): String {
        val videoLenSecond = timeToSecond(video_len)
        var posSecond = timeToSecond(pos)
        val opStartSecond = timeToSecond(op_start)
        val opEndSecond = timeToSecond(op_end)

        if (posSecond in opStartSecond..opEndSecond) {
            posSecond = opEndSecond
        }

        commands.forEach { command ->
            when (command) {
                "prev" -> {
                    if (posSecond - 10 < 0) {
                        posSecond = 0
                    } else {
                        posSecond -= 10
                    }
                    
                    if (posSecond in opStartSecond..opEndSecond) {
                        posSecond = opEndSecond
                    }
                }
                "next" -> {
                    if (posSecond + 10 > videoLenSecond) {
                        posSecond = videoLenSecond
                    } else {
                        posSecond += 10
                    }
                    
                    if (posSecond in opStartSecond..opEndSecond) {
                        posSecond = opEndSecond
                    }
                }
            }
        }

        if (posSecond in opStartSecond..opEndSecond) {
            posSecond = opEndSecond
        }

        return secondToTime(posSecond)
    }

    fun timeToSecond(time: String): Int {

        val (minute, second) = time.replace("\"", "")
            .split(":")
            .map { it.toInt() }

        return minute * 60 + second
    }

    fun secondToTime(second: Int): String {

        val minute = second / 60
        val second = second % 60

        return String.format("%02d:%02d", minute, second)
    }

}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int con_ti(string time) {
    int m = (time[0] - '0') * 10 + (time[1] - '0');
    int s = (time[3] - '0') * 10 + (time[4] - '0');
    return m * 60 + s;
}

string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {
    string answer = "";
    int pos_m = (pos[0] - '0') * 10 + (pos[1] - '0');
    int pos_s = (pos[3] - '0') * 10 + (pos[4] - '0');
    int pt = con_ti(pos);

    int op_start_m = (op_start[0] - '0') * 10 + (op_start[1] - '0');
    int op_start_s = (op_start[3] - '0') * 10 + (op_start[4] - '0');
    int ost = con_ti(op_start);

    int op_end_m = (op_end[0] - '0') * 10 + (op_end[1] - '0');
    int op_end_s = (op_end[3] - '0') * 10 + (op_end[4] - '0');
    int oet = con_ti(op_end);

    int video_len_m = (video_len[0] - '0') * 10 + (video_len[1] - '0');
    int video_len_s = (video_len[3] - '0') * 10 + (video_len[4] - '0');
    int vt = con_ti(video_len);
    
    for(int i=0; i<commands.size(); i++) {
        // if(pos_m >= op_start_m && pos_m <= op_end_m) {
        //     if(op_start_m == op_end_m) {
        //         if(pos_s <= op_end_s && pos_s >= op_start_s) {
        //             pos_m = op_end_m;
        //             pos_s = op_end_s;
        //         }   
        //     } else {
        //         if(pos_s >= op_start_s && pos_m < op_end_m) {
        //             pos_m = op_end_m;
        //             pos_s = op_end_s;
        //         } 
        //         if(pos_m == op_end_m && pos_s <= op_end_s){
        //             pos_m = op_end_m;
        //             pos_s = op_end_s;
        //         }
        //     }
        // }
        if(pt >= ost && pt <= oet) pt = oet;
        
        if(commands[i] == "next") {
            pos_s += 10;
            if(pos_s >= 60) {
                pos_s -= 60;
                pos_m += 1;
            }
            pt += 10;
        }
        if(commands[i] == "prev") {
            pos_s -= 10;
            if(pos_s < 0) {
                pos_s += 60;
                pos_m -= 1;
            }
            pt -= 10;
        }

        if(pos_m < 0) {
            pos_m = 0;
            pos_s = 0;
        }
        if(pt < 0) pt = 0;

        if(pos_m > video_len_m || (pos_m == video_len_m) && (pos_s > video_len_s)) {
            pos_m = video_len_m;
            pos_s = video_len_s;
        }
        if(pt > vt) pt = vt;
    }
    
    // if(pos_m >= op_start_m && pos_m <= op_end_m) {
    //     if(op_start_m == op_end_m) {
    //         if(pos_s <= op_end_s && pos_s >= op_start_s) {
    //             pos_m = op_end_m;
    //             pos_s = op_end_s;
    //         }   
    //     } else {
    //         if(pos_s >= op_start_s && pos_m < op_end_m) {
    //             pos_m = op_end_m;
    //             pos_s = op_end_s;
    //         } 
    //         if(pos_m == op_end_m && pos_s <= op_end_s){
    //             pos_m = op_end_m;
    //             pos_s = op_end_s;
    //         }
    //     }
    // }
    if(pt >= ost && pt <= oet) pt = oet;
    
    int mm = pt / 60;
    int ss = pt % 60;
    
    string result = "";
    if(pos_m < 10) {
        answer += "0";
    }
    if(mm < 10) {
        result += "0";
    }
    
    answer += to_string(pos_m);
    result += to_string(mm);
    answer += ":";
    result += ":";
    if(pos_s < 10) {
        answer += "0";
    }
    if(ss < 10) {
        result += "0";
    }
    
    answer += to_string(pos_s);
    result += to_string(ss);
    return result;
}